cmds/core/wc/wc_test.go

```diff
@@ -1,96 +1,144 @@
+			name: "count words in file1",
+			args: []string{f1.Name()},
+			p:    params{words: true},
+			want: fmt.Sprintf("6 %s\n", f1.Name()),
 		},
 		{
-			name:  "count lines in file1",
-			args:  []string{"testfiles/file1"},
-			lines: true,
-			want:  "3 testfiles/file1\n",
+			name: "count lines in file1",
+			args: []string{f1.Name()},
+			p:    params{lines: true},
+			want: fmt.Sprintf("3 %s\n", f1.Name()),
 		},
 		{
-			name:  "count lines in file1",
-			args:  []string{"testfiles/file1"},
-			chars: true,
-			want:  "36 testfiles/file1\n",
+			name: "count lines in file1",
+			args: []string{f1.Name()},
+			p:    params{chars: true},
+			want: fmt.Sprintf("36 %s\n", f1.Name()),
 		},
 		{
-			name:  "count runes in file1",
-			args:  []string{"testfiles/file1"},
-			runes: true,
-			want:  "36 testfiles/file1\n",
+			name: "count runes in file1",
+			args: []string{f1.Name()},
+			p:    params{runes: true},
+			want: fmt.Sprintf("36 %s\n", f1.Name()),
 		},
 		{
-			name:   "count broken in file1",
-			args:   []string{"testfiles/file1"},
-			broken: true,
-			want:   "0 testfiles/file1\n",
+			name: "count broken in file1",
+			args: []string{f1.Name()},
+			p:    params{broken: true},
+			want: fmt.Sprintf("0 %s\n", f1.Name()),
 		},
 		{
 			name: "count in 2 files",
-			args: []string{"testfiles/file1", "testfiles/file2"},
-			want: "3 6 36 testfiles/file1\n0 0 0 testfiles/file2\n3 6 36 total\n",
+			args: []string{f1.Name(), f2.Name()},
+			want: fmt.Sprintf("3 6 36 %s\n0 0 0 %s\n3 6 36 total\n", f1.Name(), f2.Name()),
 		},
-	} {
-		*lines = tt.lines
-		*words = tt.words
-		*runes = tt.runes
-		*broken = tt.broken
-		*chars = tt.chars
-		t.Run(tt.name, func(t *testing.T) {
-			bufOut := &bytes.Buffer{}
-			bufIn := &bytes.Buffer{}
-			if tt.name == "no flags no file" {
-				bufIn.Write([]byte{0x41, 0x42, 0x43, 0x44, 0x45, 0x0a})
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			stdout := &bytes.Buffer{}
+			stderr := &bytes.Buffer{}
+			if err := command(nil, stdout, stderr, test.p, test.args).run(); err != nil {
+				t.Fatal(err)
 			}
-			if got := runwc(bufOut, bufIn, tt.args...); got != nil {
-				if got.Error() != tt.wantErr {
-					t.Errorf("runwc() = %q, want: %q", got.Error(), tt.wantErr)
-				}
-			} else {
-				if bufOut.String() != tt.want {
-					t.Errorf("runwc() = %q, want: %q", bufOut.String(), tt.want)
-				}
+			if stdout.String() != test.want {
+				t.Errorf("wc stdout = %q, want: %q", stdout.String(), test.want)
+			}
+			if stderr.String() != test.wantStderr {
+				t.Errorf("wc stderr = %q, want: %q", stderr.String(), test.wantStderr)
 			}
 		})
 	}
 }
+
+func TestStdin(t *testing.T) {
+	tests := []struct {
+		input string
+		want  string
+		args  []string
+		p     params
+	}{
+		{
+			input: "hello world\n",
+			want:  "1 2 12\n",
+		},
+		{
+			input: "München\n",
+			want:  "1 1 9\n",
+		},
+		{
+			input: "",
+			want:  "0 0 0\n",
+		},
+	}
+
+	for _, test := range tests {
```

Comment:
should we use t.Run?

Reply:
I think it worth to add at least for symmetry :)